src/Config.js

```diff
@@ -86,6 +86,7 @@ export class Config {
     logLevels,
     rateLimit,
     databaseOptions,
+    renewSessions,
```

Comment:
I haven't looked much into the new auth adapter, but would it make more sense to add such a mechanism to the auth adapter instead of Parse Server? I imagine that different auth adapters may want to treat this differently. For example, a developer may want to renew session for login with adapter X, but not with adapter Y, or tie renewal to conditions. cc @Moumouls

Reply:
It could be useful as a seperate feature to allow auth adapters to specify session expiry separately and provide their own renew method (for example facebook auth), but this feature here is relating to a global session extension mechansim

Reply:
- username and password login is still hard baked into Parse Server and is not considered a seperate auth adapter. Auth adapter overhaul was just to support custom auth create, update, login and signup (whereas previously it was just `validateAuthData`). I think Username/email/password auth is considered seperate to additional auth adapters.
- Whenever Parse Server is called, a session lookup happens, regardless of the auth method that was initially used. The initial auth still creates a Parse Session and uses that session token in future.

I might be mistaken but my understanding is that auth adapter have no control over the session object that is created thereafter. If the auth data initially is valid, a session is created with `sessionLength` irrespective of the auth's actual expiry. It seems though whenever the user object is saved, the stored auth data is validated again, and if it's expired `invalid auth data` will throw.
 
We might need to allow auth adapters to control the created session expiry, but I think that's out of scope of this issue.

Reply:
Alright, then the only thing I'd recommend it to rename the option `extendSessionOnLogin` to more accurately self-describe it.

Reply:
The sessions are extended whenever the session token is used, not just on login

Reply:
Got it, maybe rename it to `extendSessionOnUse`?

- Renewing a session sounds to me like creating a nee session token, which is not the case, this extends the date of the existing token, right?
- A more specific option name is less likely to conflict with future options, like `extendSessionOnXyz`.